Add Ansible playbook for st2express

enykeev opened the pull request.

Commit:
Add Ansible playbook for st2express

Kami commented:
Nice!

Kami commented:
This is just related to the code organization - I think it would make a lot of sense to store this code in a separate repository (e.g. `ansible-playbook-st2`) and then include it here using git subtree (or submodule).

Kami commented on ansible/roles/mistral/vars/main.yml:
```diff
@@ -0,0 +1,9 @@
+mistral_branches:
```
I wish we could provide this information in some common place so it doesn't need to be hard-coded.

Or ideally, we would just assume, that going forward, there should be a branch for each version which follows a standard naming format so we don't need ifs anymore (we don't support versions older than 0.8 anymore anyway).

enykeev commented:
Ok, how about that?

I've moved playbooks and roles into [stackstorm/ansible-st2](https://github.com/StackStorm/ansible-st2) repo and changed provisioning method so it would not require user to have ansible installed on a host.

I've also made it so the repo would be automatically cloned on every new box and that it would try to keep master up-to-date with the latest changes in the origin on every provisioning while at the same time keeping all the local changes made to any other branch intact. The only situation it would ask for human interaction is when it wont be able to make fast-forward fetch either because someone has tempered with `ansible-st2` master history or because user committed some changes into local master. In both cases it is impossible to automate the procedure without knowing ones intentions.

jfryman commented on stacks/st2.yaml:
```diff
@@ -17,6 +17,8 @@ st2dev:
 st2express:
   <<: *defaults
   hostname: st2express
+  ansible:
```
Please move this example to `template.yaml`

enykeev replied:
Just to be clear, `mv` or `cp`? =)

jfryman replied:
`mv`. :)

jfryman commented on stacks/st2.yaml:
```diff
@@ -24,7 +26,8 @@ st2express:
   mounts:
     - '/opt/stackstorm:artifacts/stackstorm'
-  vm_sync: nfs
+    - '/opt/ansible:artifacts/ansible'
```
Probably should not commit this.

enykeev replied:
I've spent quite some time to make git updating properly and I wanted to make it clear you can edit and commit ansible roles from your host machine, apply the changes on a guest with `vagrant provision` and nothing would break or get lost. May be I should move it to st2dev instead or make it a part of provisioning, as you did with puppet.

Yeah, and btw, since now we have several provisioners, can we move puppet stuff into its own top level folder or there's some kind of requirements for some of the files to be in root of the project?

jfryman replied:
It's not that. It's that by default a stack shouldn't mount or assume NFS mounts. These are user configurable options, and you can/should document it in `template.yaml` for others to use. 

And as far as moving the Puppet code. Sure. It is possible to do this. Feel free to submit a PR, or I'll take a pass at it next time I sit down with this code.

jfryman commented:
@enykeev thanks for the code!
